pkg/content/local.go

```diff
@@ -130,3 +138,130 @@ func (pvd *LocalProvider) getImage(ref string) (*ocispec.Descriptor, error) {
 	}
 	return nil, errdefs.ErrNotFound
 }
+
+func (pvd *LocalProvider) FetchRemoteCache(ctx context.Context, ref string) (*Cache, error) {
+
+	resolver, err := pvd.Resolver(ref)
+	if err != nil {
+		return pvd.remoteCache, err
+	}
+
+	// TODO: sets max concurrent downloaded layer limit by containerd.WithMaxConcurrentDownloads.
+	rc := &containerd.RemoteContext{
+		Resolver:        resolver,
+		PlatformMatcher: pvd.platformMC,
+	}
+
+	name, desc, err := rc.Resolver.Resolve(ctx, ref)
+	if err != nil {
+		return pvd.remoteCache, err
+	}
+
+	fetcher, err := rc.Resolver.Fetcher(ctx, name)
+	if err != nil {
+		return pvd.remoteCache, err
+	}
+
+	//fetch cache manifest
+	ir, err := fetcher.Fetch(ctx, desc)
+	if err != nil {
+		if nydusErrdefs.NeedsRetryWithHTTP(err) {
+			pvd.UsePlainHTTP()
+			ir, err = fetcher.Fetch(ctx, desc)
+			if err != nil {
+				return nil, errors.Wrap(err, "try to pull remote cache")
+			}
+		} else {
+			return pvd.remoteCache, errors.Wrap(err, "pull remote cache")
+		}
+	}
+
+	bytes, err := io.ReadAll(ir)
+	if err != nil {
+		return pvd.remoteCache, err
+	}
+
+	manifest := ocispec.Manifest{}
+	if err = json.Unmarshal(bytes, &manifest); err != nil {
+		return pvd.remoteCache, err
+	}
+
+	pvd.remoteCache.Clear()
+	for _, layer := range manifest.Layers {
+		if layer.Annotations["containerd.io/snapshot/nydus-blob"] == "true" {
+			pvd.remoteCache.Add(layer.Annotations["source image layer"], layer)
+		}
+	}
+	return pvd.remoteCache, nil
+}
+
+// update remote cache and push to remote
```

Comment:
Please note the exported method comment format.
How about `UpdateRemoteCache updates the remote cache and push to remote`.